tooling/nargo_cli/src/cli/clean_cmd.rs

```diff
@@ -0,0 +1,276 @@
+
+    // Consider a path "within workspace" if:
+    // - It canonically resolves under the workspace root, or
+    // - It does not exist yet but lexically sits under the workspace root.
+    let within_workspace = |p: &Path| {
+        let abs = if p.is_absolute() { p.to_path_buf() } else { workspace.root_dir.join(p) };
+        match abs.canonicalize() {
+            Ok(canon) => canon.starts_with(&workspace_root),
+            Err(_) => abs.starts_with(&workspace_root),
+        }
+    };
+
+    if remove_target {
+        if !within_workspace(target_root) {
+            return Err(CliError::Generic(format!(
+                "Refusing to clean target outside workspace: {}",
+                target_root.display()
+            )));
+        }
+        planned.push(PlannedPath { label: "target".into(), path: target_root.to_path_buf() });
+    }
+
+    // Only schedule local CRS subdirectories if we are not removing the entire target.
+    if remove_crs && !remove_target {
+        for dir in ["crs", "srs"] {
+            let p = target_root.join(dir);
+            if p.is_dir() && within_workspace(&p) {
+                planned.push(PlannedPath { label: format!("local-{dir}"), path: p });
+            }
+        }
+    }
+
+    // Optionally include the global CRS directory; only with --force.
+    if remove_crs {
+        match (cmd.force, global_crs) {
+            (true, Some(p)) if p.is_dir() => {
+                planned.push(PlannedPath { label: "global-crs".into(), path: p.clone() });
+            }
+            (false, Some(p)) if p.is_dir() && cmd.verbose => {
+                println!(
+                    "Global CRS cache at {} ({}) (use --force with --crs to remove)",
+                    p.display(),
+                    dir_size_string(p)
+                );
+            }
+            (true, None) if cmd.verbose => println!("(No global CRS cache found)"),
+            _ => {}
+        }
+    }
+
+    // Deduplicate and sort for stable output.
+    let mut seen = HashSet::new();
+    planned.retain(|e| seen.insert(canonical_key(&e.path)));
+    planned.sort_by_key(|e| (e.label.clone(), e.path.clone()));
+
+    Ok(planned)
+}
+
+fn remove_path(p: &Path) -> io::Result<()> {
+    match fs::metadata(p) {
+        Ok(md) if md.is_dir() => fs::remove_dir_all(p),
+        Ok(_) => fs::remove_file(p),
+        Err(e) => Err(e),
+    }
+}
+
+fn canonical_key(p: &Path) -> String {
+    p.canonicalize()
+        .map(|c| c.to_string_lossy().into_owned())
+        .unwrap_or_else(|_| p.to_string_lossy().into_owned())
+}
+
+fn possible_global_crs_dir() -> Option<PathBuf> {
+    if let Ok(env_path) = std::env::var("NARGO_GLOBAL_CRS_DIR") {
+        let p = PathBuf::from(env_path);
+        if p.is_dir() {
+            return Some(p);
+        }
+    }
+    let home = home_dir()?;
+    [home.join(".nargo/crs"), home.join(".cache/nargo/crs")].into_iter().find(|p| p.is_dir())
```

Comment:
In general the CRS is managed by the backend rather than by `nargo`. You are not going to find a general "CRS" saved in these locations. We should remove this logic.